File: src/main/kotlin/racingcar/domain/MoveStrategy.kt
package racingcar.domain

import racingcar.domain.moverule.MoveRule
import racingcar.domain.numbergenerator.NumberGenerator

class MoveStrategy(
    private val moveRule: MoveRule,
    private val numberGenerator: NumberGenerator,
) {
    fun canMove(): Boolean {
        val number = numberGenerator.getNumber()
        return moveRule.canMove(number)
    }
}

File: racingcar/racingcar/domain/MoveStrategy.swift
//
//  MoveStrategy.swift
//  racingcar
//
//  Created by KimMinSeok on 4/3/24.
//

import Foundation

class MoveStrategy{
    private let numberGenerator: NumberGeneator
    private let moveRule : MoveRule
    
    init(numberGenerator : NumberGeneator, moveRule : MoveRule){
        self.numberGenerator = numberGenerator
        self.moveRule = moveRule
    }
    
    func canMove() -> Bool{
        return moveRule.canMove(number: numberGenerator.getNumber())
    }
    
}
